### PR TITLE
feat: [PIE-8191]: add parentStageInfo in recent executions info of child pipeline on pipelines list page

### PR DESCRIPTION
**Pull request by shalini-agr**
## Describe your changes
**Issue:** In the pipelines list page, if a pipeline is triggered through a cron or webhook trigger, then in the execution of its child pipeline, the executor's info does not show up in UI and it just shows 'this pipeline never ran'. 

Screenshot 2023-02-22 at 10 34 10 AM

**Changes:** Add parentStageInfo in recent executions info of child pipeline on pipelines list page.

**Impact:** Now, the parent pipeline's name will be displayed in the executor's info of the child pipeline just like below.

Screenshot 2023-02-22 at 10 35 28 AM

**Testing:**
Verified that parentStageInfo is getting populated in the recentExecutionInfo of the child pipeline. Tested for both normal and v1  APIs. Also checked that correct values are still populated for pipelines not having a parent stage.

Screenshot 2023-02-21 at 7 51 41 PM

The 0th element in below list is the execution after the changes, and 1st element is before

Screenshot 2023-02-21 at 5 46 57 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)

### CODE REVIEW

**Review by prashantsharma04 — DISMISSED** [review]
> Reviewed 3 of 4 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 3 of 4 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: 1 change requests. 1 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1014 at r2](https://reviewable.io/reviews/harness/harness-core/44163#-NOs-kim2AMNYr3q2gmK:-NOs-kim2AMNYr3q2gmL:b-lakxwe) ([raw file](https://github.com/harness/harness-core/blob/dc9eb19a1b270c0a52edabfcd42c6a516cacf272/pipeline-service/contracts/openapi/v1/openapi.yaml#L1014)):*
> ```YAML
>           description: Execution identifier
>         started:
>           type: integer
> ```

get api squad approval as well check if any documentation update is required?, why we removed started?
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1049 at r2](https://reviewable.io/reviews/harness/harness-core/44163#-NOs-tsn5hVmlQ8-LNDZ:-NOs-tsn5hVmlQ8-LND_:b-lp245s) ([raw file](https://github.com/harness/harness-core/blob/578ba041923e6cadb0e63c7bed93334968b8766f/pipeline-service/contracts/openapi/v1/openapi.yaml#L1049)):*
> ```YAML
>         example-recent:
>           executor_info:
>             trigger_type: NOOP
> ```

why this is removed?

**Review by mankrit-singh — DISMISSED** [review]
> Reviewed 3 of 4 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: :shipit: complete! ✓

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -999,6 +1010,10 @@ components:
           minLength: 1
           maxLength: 128
           description: Execution identifier
+        started:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:1013]: nit, please move the "started" parameter back to original positon
Moving around examples / this parameter wont affect the functionality, but might moving this parameter would affect the documentation of API

**Comment by shalini-agr** [comment]
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1013 at r2](https://reviewable.io/reviews/harness/harness-core/44163#-NOoGpHsoOqhFcEcA6Gg-r2-1013:-NOsDSmg7dMAmNVmApBK:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/dc9eb19a1b270c0a52edabfcd42c6a516cacf272/pipeline-service/contracts/openapi/v1/openapi.yaml#L1013)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

nit, please move the "started" parameter back to original positon
Moving around examples / this parameter wont affect the functionality, but moving this parameter would affect the documentation of API
</blockquote></details>

done

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1014 at r2](https://reviewable.io/reviews/harness/harness-core/44163#-NOs-kim2AMNYr3q2gmK:-NOsDUqHBolgku6MHbfE:bpdzn64) ([raw file](https://github.com/harness/harness-core/blob/dc9eb19a1b270c0a52edabfcd42c6a516cacf272/pipeline-service/contracts/openapi/v1/openapi.yaml#L1014)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

get api squad approval as well check if any documentation update is required?, why we removed started?
</blockquote></details>

this was refactored automatically when i generated using stoplight studio, have manually reverted them now.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1049 at r2](https://reviewable.io/reviews/harness/harness-core/44163#-NOs-tsn5hVmlQ8-LNDZ:-NOsDfdV9lSi99QDCeGT:bpdzn64) ([raw file](https://github.com/harness/harness-core/blob/578ba041923e6cadb0e63c7bed93334968b8766f/pipeline-service/contracts/openapi/v1/openapi.yaml#L1049)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why this is removed?
</blockquote></details>

this was refactored automatically when i generated using stoplight studio, have manually reverted them now.

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by prashantsharma04 — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: 1 change requests. 2. unresolved discussions.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44163)* status: :shipit: complete! ✓

**Pull request merged**